BadRequest: 400 null - missing componentType when creating non existent component from release

### Summary of the Bug
Run Example Project on empty SW360 Db instance

```
Caused by: org.springframework.web.client.HttpClientErrorException$BadRequest: 400 null
    at org.springframework.web.client.HttpClientErrorException.create (HttpClientErrorException.java:79)
    at org.springframework.web.client.DefaultResponseErrorHandler.handleError (DefaultResponseErrorHandler.java:97)
    at org.springframework.web.client.DefaultResponseErrorHandler.handleError (DefaultResponseErrorHandler.java:79)
    at org.springframework.web.client.ResponseErrorHandler.handleError (ResponseErrorHandler.java:63)
    at org.springframework.web.client.RestTemplate.handleResponse (RestTemplate.java:777)
    at org.springframework.web.client.RestTemplate.doExecute (RestTemplate.java:735)
    at org.springframework.web.client.RestTemplate.execute (RestTemplate.java:669)
    at org.springframework.web.client.RestTemplate.exchange (RestTemplate.java:607)
    at org.eclipse.sw360.antenna.sw360.rest.SW360Client.doRestCall (SW360Client.java:58)
    at org.eclipse.sw360.antenna.sw360.rest.SW360Client.doRestPOST (SW360Client.java:70)
    at org.eclipse.sw360.antenna.sw360.rest.SW360ComponentClient.createComponent (SW360ComponentClient.java:85)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ComponentClientAdapter.createComponent (SW360ComponentClientAdapter.java:47)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ComponentClientAdapter.lambda$getOrCreateComponent$0 (SW360ComponentClientAdapter.java:40)
    at java.util.Optional.orElseGet (Optional.java:267)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ComponentClientAdapter.getOrCreateComponent (SW360ComponentClientAdapter.java:40)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ReleaseClientAdapter.createRelease (SW360ReleaseClientAdapter.java:71)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ReleaseClientAdapter.lambda$getOrCreateRelease$1 (SW360ReleaseClientAdapter.java:55)
    at java.util.Optional.orElseGet (Optional.java:267)
    at org.eclipse.sw360.antenna.sw360.adapter.SW360ReleaseClientAdapter.getOrCreateRelease (SW360ReleaseClientAdapter.java:55)
    at org.eclipse.sw360.antenna.sw360.SW360MetaDataUpdater.getOrCreateRelease (SW360MetaDataUpdater.java:72)
    at org.eclipse.sw360.antenna.sw360.workflow.generators.SW360UpdaterImpl.produce (SW360UpdaterImpl.java:52)
    at org.eclipse.sw360.antenna.sw360.workflow.generators.SW360Updater.produce (SW360Updater.java:62)

```

### Steps to reproduce
- Run Example Project on empty SW360 Db instance
- OR 
- make up new release with only name and version as info

### Acceptance Criteria
- [ ] Make request for createComponent Valid
- [ ] no error occurs

### Definition of Done
- Acceptance criteria fulfilled
- A test case is created to reproduce the bug
- A PR is created, the CI infrastructure reports green, the bug test case proves that bug is fixed
- The PR is reviewed and approved
- No TODOs left in the code unless explained in the ticket, if something else is still open, this is summarized in a comment in the issue
- Documentation is updated

--- a/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java
+++ b/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java
@@ -36,6 +36,8 @@ public class SW360Release extends SW360HalResource<SW360ReleaseLinkObjects, SW36
     private static final String COPYRIGHTS_KEY = "copyrights";
     private static final String CLEARINGSTATE_KEY = "clearingState";
 
+    @JsonIgnore
+    private boolean isProprietary;
     private String name;
     private String version;
     private String cpeId;
@@ -263,6 +265,17 @@ public class SW360Release extends SW360HalResource<SW360ReleaseLinkObjects, SW36
 
     public SW360Release setCopyrights(String copyrights) {
         additionalData.put(COPYRIGHTS_KEY, copyrights);
+        return this;
+    }
+
+    @JsonIgnore
+    public boolean isProprietary() {
+        return isProprietary;
+    }
+
+    @JsonIgnore
+    public SW360Release setProprietary(boolean proprietary) {
+        isProprietary = proprietary;
         return this;
     }
 

--- a/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ComponentAdapterUtils.java
+++ b/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ComponentAdapterUtils.java
@@ -37,7 +37,11 @@ public class SW360ComponentAdapterUtils {
     }
 
     private static void setComponentType(SW360Component component, Artifact artifact) {
-        if (artifact.isProprietary()) {
+        setComponentType(component, artifact.isProprietary());
+    }
+
+    private static void setComponentType(SW360Component component, boolean isProprietary) {
+        if (isProprietary) {
             component.setComponentType(SW360ComponentType.INTERNAL);
         } else {
             component.setComponentType(SW360ComponentType.OSS);
@@ -47,7 +51,7 @@ public class SW360ComponentAdapterUtils {
     public static SW360Component createFromRelease(SW360Release release) {
         SW360Component sw360Component = new SW360Component();
         sw360Component.setName(release.getName());
-        // TODO: set component type
+        setComponentType(sw360Component, release.isProprietary());
         return sw360Component;
     }
 

--- a/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ReleaseAdapterUtils.java
+++ b/modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ReleaseAdapterUtils.java
@@ -39,6 +39,7 @@ public class SW360ReleaseAdapterUtils {
         SW360ReleaseAdapterUtils.setClearingStatus(release, artifact);
         SW360ReleaseAdapterUtils.setChangeStatus(release, artifact);
         SW360ReleaseAdapterUtils.setCopyrights(release, artifact);
+        release.setProprietary(artifact.isProprietary());
 
         return release;
     }

